plugin/pkg/scheduler/scheduler.go

```diff
@@ -92,6 +97,12 @@ func (s *Scheduler) scheduleOne() {
 		glog.V(1).Infof("Failed to schedule: %+v", pod)
 		s.config.Error(pod, err)
 		s.config.Recorder.Eventf(pod, api.EventTypeWarning, "FailedScheduling", "%v", err)
+		s.config.PodConditionUpdater.Update(pod, &api.PodCondition{
```

Comment:
I think @hongchaodeng has a good point. Is this effectively a synchronous call to client to write to API server? That seems like a bad thing to put in the scheduling loop. Making the update asynchronous would be one approach. Another approach would be to have UpdatePodCondition() only write new pod status if the condition changed, so this would only be called synchronously once for each time a pod goes pending. So you'd have UpdatePodCondition() return a bool indicating whether condition changed; only if it returns true would you call UpdateStatus().